spanner/tests/system/test_system.py

```diff
@@ -580,6 +581,30 @@ def test_read_w_index(self):
             [(row[0], row[2]) for row in self._row_data(ROW_COUNT)]))
         self._check_row_data(rows, expected)
 
+    def test_read_w_single_key(self):
+        ROW_COUNT = 40
+        session, committed = self._set_up_table(ROW_COUNT)
+
+        snapshot = session.snapshot(read_timestamp=committed)
+        rows = list(snapshot.read(
+            self.TABLE, self.COLUMNS, KeySet(keys=[(0,)])))
+
+        all_data_rows = list(self._row_data(ROW_COUNT))
+        expected = all_data_rows[:1]
```

Comment:
This confused me at first (i.e. I wondered why you had `all_data_rows[:1]`). Will be clearer if you did something like:

```python
indices = [0]
key_set = KeySet([(index,) for index in indices])
...
expected = [all_data_rows[index] for index in indices]
```